examples/echo/main.c

```diff
@@ -33,6 +33,73 @@ g_date_time_format_iso8601 (GDateTime *datetime)
 }
 #endif
 
+/** sleep_delay:
+ *
+ * The number of seconds the handler will sleep before execution.
+ */
+static gint sleep_delay = 0;
+
+/** loop_times:
+ *
+ * The number of times the response handler is invoked.
+ */
+static gint loop_times = 1;
+
+/** loops:
+ *
+ * The number of times the response handler **has been** invoked.
+ */
+static gint loops = 1;
+
+static GHashTable *active_sources;
+
+typedef struct _RxData
+{
+  YggWorker    *worker;
+  gchar        *addr;
+  gchar        *id;
+  gchar        *response_to;
+  YggMetadata  *metadata;
+  GBytes       *data;
+  GCancellable *cancellable;
+} _RxData;
```

Comment:
not to be ultra-pedantic: identifiers starting with underscore are considered reserved for the implementation and can appear in the global namespace any time

what about naming it `EchoRxData`, `Echo_RxData`, or something like that?

Reply:
Ultra-pedantic is good; C demands it. The global namespace here is going to be the global namespace of the C echo worker though, correct? This file is not included in `libygg.so`, so `_RxData` is not a symbol in that library; only in the echo worker program.

Reply:
> The global namespace here is going to be the global namespace of the C echo worker though, correct?

True, however that also means everything you get in each translation unit via `#include`, system headers included (pun semi-intended for lack of good synonyms here).

So a system header may potentially eg `#define _RxData <something>` -- even if this actual name is very unlike, you get my idea here.

Reply:
Ah, I was considering it from the other side (our symbol clobbering another's), but I didn't consider someone clobbering our symbol. This makes more sense to scope it within the namespace of the echo worker.